crates/iota-indexer/tests/rpc-tests/indexer_api.rs

```diff
@@ -751,3 +752,195 @@ fn test_get_dynamic_field_objects() -> Result<(), anyhow::Error> {
         Ok(())
     })
 }
+
+#[test]
+fn test_query_transaction_blocks_tx_kind_filter() -> Result<(), anyhow::Error> {
+    let ApiTestSetup {
+        runtime,
+        store,
+        cluster,
+        client,
+    } = ApiTestSetup::get_or_init();
+
+    runtime.block_on(async move {
+        let (address, keypair): (_, AccountKeyPair) = get_key_pair();
+
+        let gas = cluster
+            .fund_address_and_return_gas(
+                cluster.get_reference_gas_price().await,
+                Some(500_000_000),
+                address,
+            )
+            .await;
+        let iota_client = cluster.wallet.get_client().await.unwrap();
+
+        indexer_wait_for_object(client, gas.0, gas.1).await;
+
+        let objects = client
+            .get_owned_objects(
+                address,
+                Some(IotaObjectResponseQuery::new_with_options(
+                    IotaObjectDataOptions::new()
+                        .with_type()
+                        .with_owner()
+                        .with_previous_transaction(),
+                )),
+                None,
+                None,
+            )
+            .await?
+            .data;
+
+        assert_eq!(objects.len(), 1);
+
+        let signer = address;
+
+        let package_id = MOVE_STDLIB_PACKAGE_ID;
+        let module = Identifier::from_str("address")?;
+        let function = Identifier::from_str("length")?;
+
+        let mut pt_builder = ProgrammableTransactionBuilder::new();
+        pt_builder.move_call(package_id, module, function, vec![], vec![])?;
+        let pt = pt_builder.finish();
+
+        let tx_data = TransactionData::new_programmable(signer, vec![gas], pt, 10_000_000, 1_000);
+        let signed_transaction = to_sender_signed_transaction(tx_data, &keypair);
+
+        let response = iota_client
+            .quorum_driver_api()
+            .execute_transaction_block(
+                signed_transaction,
+                IotaTransactionBlockResponseOptions::new(),
+                Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+            )
+            .await
+            .unwrap();
+
+        indexer_wait_for_transaction(response.digest, store, client).await;
+
+        let options = IotaTransactionBlockResponseOptions::new().with_input();
+
+        // Test `ProgrammableTransaction` transaction kind filter
+        let filter =
+            TransactionFilter::TransactionKind(IotaTransactionKind::ProgrammableTransaction);
+        let query = IotaTransactionBlockResponseQuery::new(Some(filter), Some(options.clone()));
+        let res = client
+            .query_transaction_blocks(query, None, Some(1), Some(true))
+            .await
+            .unwrap();
+        assert_eq!(1, res.data.len());
+        //
```

Comment:
are these empty comment lines intended?

Reply:
Fixed in https://github.com/iotaledger/iota/pull/5801/commits/9e31a7bdb5047d1513b53116e88e40ec027d6b30